src/node_modules/app/elasticsearch/config.spec.js

```diff
@@ -0,0 +1,48 @@
+import assert from 'assert';
+import * as _ from 'lamb';
+import { docsByParamByAgg, paramsByAggs } from './config';
+
+const evaluate = _.when(_.isType('Function'), _.applyTo(['dummyType']));
+
+const process = _.pipe([
+	_.pairs,
+	_.mapWith(([key, definition]) => [key, evaluate(definition)]),
+]);
+
+const getParams = _.mapWith(([key, definition]) => [key, _.keys(definition)]);
+
+const excludeParams = params => _.pipe([
+	_.mapWith(([key, definition]) => [key, _.pull(params)(definition)])
+]);
+const pluckParams = params => _.filterWith(agg => _.isIn(agg[1], params));
```

Comment:
```suggestion
const pluckParams = params => _.filterWith(_.pipe([_.getAt(1), _.contains(params)]));
```

I'd probably give `_.pipe([_.getAt(1), _.contains(params)])` a name so that we'd read `pluckParams = params => _.filterWith(aCriteriaBasedOn(params))`

Also, not sure I'd use "pluck" as in Lamb that's `key => _.mapWith(_.getKey(key))`, it might be confusing.